### DISCUSSION TITLE
Add socketsupply into existing typescript frontend app?

### DISCUSSION THREAD

**Discussion by Diomas** [socketsupply/socket#908] [Q&A]
Hello everyone! I'm wondering if it's possible to add socketsupply into existing fromtend-project to make it a desktop app. That project is typescript frontend-only and is being bootstrapped with vite. Any ideas how can I combine these separate toolchains to work together?

**Comment by rmcdonald1442** [comment]
Hey, here is an example of vite being used with socket-supply, with dev mode working / hot module reloading (haven't tested the dev mode thoroughly): https://github.com/rorymcd98/socket-supply-vite-dev-bug

Despite the word "bug" in the name, this works for me on Linux, but doesn't on window (the windows part being the bug).

**Reply by rmcdonald1442** [reply]
@heapwolf are there type declarations available for the various runtime apis?

**Reply by jwerle** [reply]
Yep right here https://github.com/socketsupply/socket/blob/master/api/index.d.ts
